sycl/test-e2e/ESIMD/unified_memory_api/Inputs/atomic_update.hpp

```diff
@@ -0,0 +1,364 @@
+    T gold = init(i, cfg);
+    return gold;
+  }
+};
+
+template <class T, int N>
+struct ImplLoad : ImplLoadBase<T, N, atomic_op, atomic_op::load> {};
+template <class T, int N>
+struct ImplInc : ImplIncBase<T, N, atomic_op, atomic_op::inc> {};
+template <class T, int N>
+struct ImplDec : ImplDecBase<T, N, atomic_op, atomic_op::dec> {};
+
+// ----------------- Main function and test combinations.
+
+template <int N, template <class, int> class Op, bool UseProperties,
+          int SignMask = (Signed | Unsigned)>
+bool test_int_types(queue q, const Config &cfg) {
+  bool passed = true;
+  if constexpr (SignMask & Signed) {
+    // Supported by LSC atomic:
+    if constexpr (UseProperties)
+      passed &= test<int16_t, N, Op, UseProperties>(q, cfg);
+
+    passed &= test<int32_t, N, Op, UseProperties>(q, cfg);
+  }
+
+  if constexpr (SignMask & Unsigned) {
+    // Supported by LSC atomic:
+    if constexpr (UseProperties)
+      passed &= test<uint16_t, N, Op, UseProperties>(q, cfg);
+
+    passed &= test<uint32_t, N, Op, UseProperties>(q, cfg);
+  }
+  return passed;
+}
+
+template <int N, template <class, int> class Op, bool UseProperties>
+bool test_fp_types(queue q, const Config &cfg) {
+  bool passed = true;
+  passed &= test<float, N, Op, UseProperties>(q, cfg);
+  if (q.get_device().has(sycl::aspect::atomic64) &&
+      q.get_device().has(sycl::aspect::fp64)) {
+    // Disable double data type for fcmpxchg operation as D64 data is not
+    // supported for that operation.
+    passed &= test<double, N, Op, UseProperties>(q, cfg);
+  }
+  return passed;
+}
+
+template <template <class, int> class Op, bool UseProperties,
+          int SignMask = (Signed | Unsigned)>
+bool test_int_types_and_sizes(queue q, const Config &cfg) {
+  bool passed = true;
+
+  passed &= test_int_types<1, Op, UseProperties, SignMask>(q, cfg);
+  passed &= test_int_types<2, Op, UseProperties, SignMask>(q, cfg);
+  passed &= test_int_types<4, Op, UseProperties, SignMask>(q, cfg);
+
+  passed &= test_int_types<8, Op, UseProperties, SignMask>(q, cfg);
+
+  // Supported by LSC atomic:
+  if constexpr (UseProperties) {
+    passed &= test_int_types<16, Op, UseProperties, SignMask>(q, cfg);
+    passed &= test_int_types<32, Op, UseProperties, SignMask>(q, cfg);
+  }
+
+  return passed;
+}
+
+template <template <class, int> class Op, bool UseProperties>
+bool test_fp_types_and_sizes(queue q, const Config &cfg) {
+  bool passed = true;
+
+  passed &= test_fp_types<1, Op, UseProperties>(q, cfg);
+  passed &= test_fp_types<2, Op, UseProperties>(q, cfg);
+  passed &= test_fp_types<4, Op, UseProperties>(q, cfg);
+
+  passed &= test_fp_types<8, Op, UseProperties>(q, cfg);
+  // Supported by LSC atomic:
+  if constexpr (UseProperties) {
+    passed &= test_fp_types<16, Op, UseProperties>(q, cfg);
+    passed &= test_fp_types<32, Op, UseProperties>(q, cfg);
+  }
+  return passed;
+}
+
+template <bool UseProperties> bool test_main(queue q) {
```

Comment:
UseProperties passed to here is not used.

Reply:
Thanks, done in https://github.com/intel/llvm/pull/11833/commits/ffe19702c45355afd8b7ab17d2170cd783e9f670.